wpilibj/src/main/java/edu/wpi/first/wpilibj/XboxController.java

```diff
@@ -15,6 +15,63 @@
  * and the mapping of ports to hardware buttons depends on the code in the Driver Station.
  */
 public class XboxController extends GenericHID {
+  /** Which hand/side the of the controller to read from. */
+  public enum Hand {
+    kLeft(0),
+    kRight(1);
+
+    public final int value;
+
+    Hand(int value) {
+      this.value = value;
+    }
+  }
+
+  private final MappingPreset m_mapping;
+
+  public enum MappingPreset {
```

Comment:
I don't think this design will work in C++ because enums can't have custom constructors. Effectively, the only constructor they support is `Xbox(int i)` (note that that allows implicit conversion).

Reply:
I was thinking of making this a struct in C++. In Java, I'll change it to a fully individual class with existing static objects (similar to how the `Color` classes are set up).